api/envoy/config/route/v3/route_components.proto

```diff
@@ -300,6 +300,11 @@ message Route {
   // If set, the bytes actually buffered will be the minimum value of this and the
   // listener per_connection_buffer_limit_bytes.
   google.protobuf.UInt32Value per_request_buffer_limit_bytes = 16;
+
+  // [#not-implemented-hide:]
+  // Additional clusters which are not referenced directly by a RouteAction
+  // cluster_specifier, but may be selected by a cluster_specifier_plugin.
```

Comment:
Might be a good idea to explicitly state that this is intended for clients (like gRPC) that fetch the specific set of cluster resources referred to in the RouteConfiguration rather than making a wildcard CDS subscription.